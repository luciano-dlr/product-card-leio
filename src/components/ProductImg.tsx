import React from 'react';
import { CSSProperties, useContext } from "react";
import NoImage from "../assets/no-image.jpg";
import styles from "../styles/styles.module.css";
import ProductContext from "./ProductCardContext";


export interface ProductCardImgProps {
  className?: string;
  style?: CSSProperties;
  img?: string;
}

export const ProductImg = ({ className, style, img }: ProductCardImgProps) => {
  const { product } = useContext(ProductContext);
  // const { img } = product;


  let imgToShow: string;

  if (img) {
    imgToShow = img;
  } else if (product.img) {
    imgToShow = product.img
  } else {
    imgToShow = NoImage;
  }

  return (
    <img
      className={`${styles.productImg} ${className}`}
      src={imgToShow}
      style={style}
      alt="Product"
    />
  );
};
