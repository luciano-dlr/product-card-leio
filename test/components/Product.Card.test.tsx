import React from "react"
import renderer from "react-test-renderer"
import { ProductCard } from '../../src/components';
import { product1 } from '../data/product';

const { act } = renderer


describe('Product title and Card', () => {

    test('Debe de mostrar el componente correctamente ', () => {

        const wraper = renderer.create(
            <ProductCard product={product1} >
                {
                    () => (
                        <h1>Product card</h1>
                    )
                }
            </ProductCard>
        )

        expect(wraper.toJSON()).toMatchSnapshot();


    })
    test('Debe incrementar el contador', () => {

        const wraper = renderer.create(
            <ProductCard product={product1} >
                {
                    ({ count, increaseBy }) => (
                        <>
                            <h1>Product Card</h1>
                            <span>{count}</span>
                            <button onClick={() => increaseBy(+1)}></button>
                        </>
                    )
                }
            </ProductCard>
        )

        let tree = wraper.toJSON();
        // console.log(tree);

        expect(tree).toMatchSnapshot();

        act(() => {
            (tree as any).children[2].props.onClick();
        })

        tree = wraper.toJSON();

        expect((tree as any).children[1].children[0]).toBe('1')






    })
})