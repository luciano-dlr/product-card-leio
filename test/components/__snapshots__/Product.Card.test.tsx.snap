// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product title and Card Debe de mostrar el componente correctamente  1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product card
  </h1>
</div>
`;

exports[`Product title and Card Debe incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;
